### PR TITLE
fix(contracts): fix logic after output finalized in challenger timeout

### PR DESCRIPTION
**Pull request by kangsorang**
# Description

Related to https://github.com/chainlight-io/2023-kroma-network-audit-issues-client/issues/9 reported by theori audit.
Fixed an issue where, for a challenge to a finalized output root, 
the asserter could not take the bond even if the challenger went into a timeout state.

### CODE REVIEW

**Comment by seolaoh** [comment]
Please name the branch name starting with a verb next time ;) like `feat/collect-bonds-after-finalized`.

File: packages/contracts/contracts/L1/ValidatorPool.sol [hunk 1]
```diff
@@ -316,6 +319,16 @@ contract ValidatorPool is ReentrancyGuardUpgradeable, Semver {
         emit BondIncreased(_outputIndex, _challenger, increased);
     }
 
+    /**
+     * @notice Increases the balance of the given address, If the increased balance is at lease
```
Comment (seolaoh) [inline packages/contracts/contracts/L1/ValidatorPool.sol:323]:
```suggestion
     * @notice Increases the balance of the given address, If the increased balance is at least
```

**Comment by kangsorang** [comment]
> Maybe we need to update `challenger.go` L444-L480 in this PR. Currently asserter does not call `challengerTimeout` when output is already finalized. Wdyt?

Yes, as you said, the condition modification in the Challengers section should be removed, thank you.

File: packages/contracts/contracts/L1/ValidatorPool.sol [hunk 2]
```diff
@@ -280,19 +285,24 @@ contract ValidatorPool is ReentrancyGuardUpgradeable, Semver {
     }
 
     /**
-     * @notice Refunds the corresponding pending bond to the given output index and challenger address
+     * @notice Releases the corresponding pending bond to the given output index and from challenger address
```
Comment (seolaoh) [inline packages/contracts/contracts/L1/ValidatorPool.sol:288]:
```suggestion
     * @notice Releases the corresponding pending bond to the given output index and challenger address
```

**Comment by seolaoh** [comment]
The commit message cannot exceed 80 characters in a line :)

File: components/validator/challenger.go [hunk 3]
```diff
@@ -442,18 +442,16 @@ func (c *Challenger) handleChallenge(outputIndex *big.Int, asserter common.Addre
...
-					return
-				}
-
 				// if output is already deleted, asserter has no incentives to handle challenge any further
 				if isOutputDeleted {
 					c.log.Info("asserter: do nothing because output is already deleted", "outputIndex", outputIndex, "challenger", challenger)
 					return
 				}
-
+				// if output is already finalized and `ChallengerTimeout` status, terminate handling
```
Comment (seolaoh) [inline components/validator/challenger.go:450]:
```suggestion
				// if output is already finalized and not `ChallengerTimeout` status, terminate handling
```

**Review by seolaoh — APPROVED** [review]
LGTM

**Review by 0xHansLee — APPROVED** [review]
lgtm. Please rebase onto `dev` again.

**Review by Pangssu — APPROVED** [review]
LGTM

**Pull request merged**